### PR TITLE
WIP TGBOT-27  Create Customer repository & Bean using Set, CRUD and Tests

### PR DESCRIPTION
**Pull request by Sav4ykVladzislau**
https://schneidervs.atlassian.net/browse/TGBOT-27
WIP While working on tests

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/bean/Customer.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.Objects;
+
+public class Customer {
+    private String id;
+    private String name;
+    private String city;
+    private String phoneNumber;
+    private String email;
+
+    public Customer(String id) {
```
Comment (AlexanderYarokhovich) [inline stoservice/src/main/java/com/tutrit/stoservice/bean/Customer.java]: Write an empty constructor and a constructor with parameters.

Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/bean/Customer.java]: I do not like it

File: stoservice/src/test/java/com/tutrit/stoservice/repository/CustomerRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CustomerRepositoryTest {
+
+    CustomerRepository customerRepository;
+
+    @BeforeEach
+    void SetUp() {
+        customerRepository = new CustomerRepository();
```
Comment (schneidervs) [inline stoservice/src/test/java/com/tutrit/stoservice/repository/CustomerRepositoryTest.java:17]: add clear

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 3]
```diff
@@ -1,41 +1,67 @@
...
 
 public class CustomerRepository {
 
-    public static Set<Customer> customers = new HashSet<>();
+    public static Logger logger = LoggerFactory.getLogger(CustomerRepository.class);
+    public static HashSet<Customer> customers = new HashSet<>();
 
     public Customer createCustomer(Customer customer) {
-        return null;
+        if (customers.isEmpty() != true)
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:17]: no need to check for confusion

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 4]
```diff
@@ -1,41 +1,67 @@
...
+            if (customers.contains(customer) == false) {
+                customers.add(new Customer());
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:19]: delete "new"

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 5]
```diff
@@ -1,41 +1,67 @@
...
+            } else {
+                logger.info("Customers already exists");
+
+            }
+
+        return customer;
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:25]: maybe something other?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 6]
```diff
@@ -1,41 +1,67 @@
...
     }
 
     public Customer findCustomer(Customer customer) {
-        return null;
+        if (customers.contains(customer) == true) {
+            return customer;
+        } else {
+            logger.warn("Customer does not exist");
+        }
+        return customer;
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:34]: maybe something other?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 7]
```diff
@@ -1,41 +1,67 @@
...
+
+            }
+
+        return customer;
     }
 
     public Customer findCustomer(Customer customer) {
-        return null;
+        if (customers.contains(customer) == true) {
+            return customer;
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:30]: maybe something other?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 8]
```diff
@@ -1,41 +1,67 @@
...
         return null;
     }
 
     public Customer updateCustomer(Customer customer) {
-        return null;
+        if (customers.contains(customer) == true)
+            logger.info("This customers exists");
+        else
+            createCustomer(customer);
+        return customer;
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:46]: maybe something other?

File: stoservice/pom.xml [hunk 9]
```diff
@@ -1,41 +1,41 @@
...
-        </plugins>
-    </build>
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+		</plugins>
+	</build>
```
Comment (yurasava) [inline stoservice/pom.xml:39]: I think these changes not needed.

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+public class CustomerRepository {
+
+    public static Logger logger = LoggerFactory.getLogger(CustomerRepository.class);
+    public static HashSet<Customer> customers = new HashSet<>();
+
+    Iterator<Customer> iterationId = customers.iterator();
+
+    public Customer createCustomer(Customer customer) {
+        if (customers.contains(customer) == false) {
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:20]: Maybe change to `if (!customers.contains(customer))`

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+            customers.add(customer);
+        } else {
+            logger.info("Customers already exists");
+        }
+        return customer;
+    }
+
+    public Customer findCustomer(Customer customer) {
+        if (customers.contains(customer) == true) {
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:29]: Maybe without `==true` ?

File: stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+        return customer;
+    }
+
+    public Customer findCustomerById(Customer id) {
+
+        return id;
+    }
+
+    public Customer updateCustomer(Customer customer) {
+        if (customers.contains(customer) == true) {
```
Comment (yurasava) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/CustomerRepository.java:43]: Maybe without `==true` ?

**Review by yurasava — COMMENTED** [review]
See comments. Some methods are not implemented. Maybe this is not the final version.

**Review by schneidervs — DISMISSED** [review]

**Review by yurasava — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Review by max-java — APPROVED** [review]
could be merged as we want to close sprint

**Pull request merged**